### ISSUE TITLE
Derived class documention no longer contains related functions using protected inheritance on Doxygen trunk

### ISSUE THREAD

**Issue by scott-zhong** [doxygen/doxygen#10972]
**Describe the bug**
Derived class documention no longer contains related functions using protected inheritance on Doxygen trunk
```
/**
 * Base
 */
class Base {
};

/**
 * \relates Base
 * fun
 */
void fun();

/**
 * Derived
 */
class Derived : protected Base {
};
```

**Screenshots**
Doxygen 1.11.0:
![image](https://github.com/doxygen/doxygen/assets/32584126/0908d1d3-daf6-4778-9c2b-ea9597cd3b49)

Doxygen trunk (4a9be1303830cb78f8def8b60888cd6e23afd34d):
![image](https://github.com/doxygen/doxygen/assets/32584126/09858981-2959-4736-93c1-5bb56f34ce2a)

**To Reproduce**
[testcase.zip](https://github.com/user-attachments/files/16059697/testcase.zip)

**Expected behavior**
Related functions shows up in derived classes.

**Additional context**
Found the change that caused this by using git bisect
```
6c356988444d77aed2a045dd13e84a195d7a6433 is the first bad commit
commit 6c356988444d77aed2a045dd13e84a195d7a6433
Author: Dimitri van Heesch <doxygen@gmail.com>
Date:   Sat Jun 8 13:26:14 2024 +0200

    [Refactoring] Make MemberTypeList a class instead of enum with bit mask

 addon/doxyparse/doxyparse.cpp |  12 +-
 src/classdef.cpp              | 152 +++++++++++-----------
 src/classdef.h                |   4 +-
 src/defgen.cpp                |  84 ++++++------
 src/dotgroupcollaboration.cpp |   2 +-
 src/dotnode.cpp               |  40 +++---
 src/doxygen.cpp               |   8 +-
 src/filedef.cpp               |  62 ++++-----
 src/groupdef.cpp              | 122 ++++++++---------
 src/index.cpp                 |   2 +-
 src/layout.cpp                | 208 ++++++++++++++---------------
 src/membergroup.cpp           |   2 +-
 src/memberlist.cpp            |  66 +---------
 src/memberlist.h              |   3 +-
 src/moduledef.cpp             |  34 ++---
 src/namespacedef.cpp          |  60 ++++-----
 src/perlmodgen.cpp            |  90 ++++++-------
 src/sqlite3gen.cpp            |  10 +-
 src/trace.h                   |   8 ++
 src/types.h                   | 296 ++++++++++++++++++++++++++++--------------
 src/util.cpp                  | 191 +++++----------------------
 src/util.h                    |   4 +-
 src/vhdldocgen.cpp            |  16 +--
 src/xmlgen.cpp                |  70 ++--------
 24 files changed, 713 insertions(+), 833 deletions(-)
```

**Comment by doxygen** [comment]
@scott-zhong Please check if the referenced commit fixes the problem.

**Comment by scott-zhong** [comment]
@doxygen That fixed the problem. Thank you!!!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
